app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,74 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_member!, except: %i[index show]
```

Comment:
- [ ] Please make use of a custom user login validation method instead of authenticate_user! provided by devise.
Declare this custom method in your ApplicationController (screenshot from Odin).
Remember to restrict the before_action to "only" :new and :create actions.

You can use something like this:
````ruby
# in apps/controllers/application_controller.rb

# must be a helper method to be used in views
helper_method :signed_in?

def signed_in?
  !current_user.nil?
end

# in apps/controllers/posts_controller.rb

before_action :signed_in_only!, only: [:new, :create]
.
.
.
private

def signed_in_only!
  redirect_to new_user_session_path unless signed_in?
end
````

Reply:
Resolved using the TSE's guidance